tests/testscript.py

```diff
@@ -10,7 +10,46 @@
     from nixos_test_stubs import start_all, computeVM, controllerVM  # type: ignore
 
 
-class LibvirtTests(unittest.TestCase):
+class PrintLogsOnErrorTestCase(unittest.TestCase):
+    """
+    Custom TestCase class that prints interesting logs in error case.
+    """
+    def run(self, result=None):
+        if result is None:
+            result = self.defaultTestResult()
+
+        original_addError = result.addError
+        original_addFailure = result.addFailure
+
+        def custom_addError(test, err):
+            self.print_logs(f"Error in {test._testMethodName}")
+            original_addError(test, err)
+
+        def custom_addFailure(test, err):
+            self.print_logs(f"Failure in {test._testMethodName}")
+            original_addFailure(test, err)
+
+        result.addError = custom_addError
+        result.addFailure = custom_addFailure
+
+        return super().run(result)
+
+    def print_machine_log(self, machine, path):
```

Comment:
very nit: ```suggestion
    def print_machine_log(self, component, machine, path):
```

where `component` is `chv` or `libvirt`